tele_giphy/game/tests/test_giphy.py

```diff
@@ -7,112 +7,119 @@
+    def setUp(self, json_filename):
+        self.status = 403
+        self.json = load_json(json_filename)
+        if 'translate' in json_filename:
+            url = BASE_GIPHY_URL + "translate?api_key=abc&s=doge"
+        elif 'random' in json_filename:
+            url = BASE_GIPHY_URL + "random?api_key=abc&tag=doge"
+        else:
+            raise ValueError
+        responses.add(responses.GET, url, json=self.json,
+                      status=self.status, match_querystring=True)
 
     @responses.activate
-    def testExpectedResponse(self):
-        resp = gif_translate()
+    def test_functions(self, json_filename):
+        if 'translate' in json_filename:
+            resp = gif_translate(api_key='abc', string='doge')
+            expected = load_json("translate_{}.json".format(self.status))
+        elif 'random' in json_filename:
+            resp = gif_random(api_key='abc', tag='doge')
+            expected = load_json("random_{}.json".format(self.status))
+        else:
+            raise ValueError
         expected = self.json
         # status code
-        assert resp.status_code == 200
+        assert resp.status_code == 403
         # json is the same between fixture and request
         for item in expected:
             assert item in resp.json()
 
-
-# Tests where giphy.gif_translate get 403
-@pytest.mark.usefixtures('load_json')
-@pytest.mark.parametrize("load_json", ['translate_403.json'], indirect=True)
-class TestTranslateFail:
-    # Sets up request fixture
-    @pytest.fixture(autouse=True)
-    def setUp(self, load_json):
-        self.json = load_json
-        # GET 403 setup
-        # Might want a more comprehensive regex to capture wrong api key
-        responses.add(responses.GET,
-                      'http://api.giphy.com/v1/gifs/translate?api_key=abc&s=leeroy',
-                      json=self.json, status=403, match_querystring=True)
-
     @responses.activate
-    def testExpectedResponse(self):
-        resp = gif_translate(api_key='abc')
-        expected = self.json
-        # status code
-        assert resp.status_code == 403
-        # json is the same between fixture and request
-        for item in expected:
-            assert item in resp.json()
+    def test_funnel(self, json_filename):
+        if 'random' in json_filename:
+            resp = giphy_call(call_type='random', api_key='abc')
+            expected = {'call_type': '',
+                        'image_url': '',
+                        'phrase': '',
+                        'meta': {'status': 403, 'msg': 'Forbidden'}}
+        else:
+            resp = giphy_call(api_key='abc')
+            expected = {'call_type': '',
+                        'image_url': '',
+                        'phrase': '',
+                        'meta': {'status': 403, 'msg': 'Forbidden'}}
+        assert resp['meta']['status'] == 403
+        # Check standarized dict
+        assert expected == resp
```

Comment:
Names overall in this module aren't the best. Most names are non-descriptive (and possibly confusing). 
- TestFail is especially contradictory. Those tests aren't failing, they are passing. (This wouldn't be as bad if other names were more descriptive, but none are.)
- test_functions, what functions? All functions? Then why not test_module (which is also a bad name)? Should it be changed if more functions are added? etc etc
- test_funnel, what is funnel? Nowhere can one see anything that is called funnel.

There's some duplication that could easily be refactored out.

Note that you don't have to explain any of these things to me. I understand all of it since I've been here from the start and know how you're thinking. just pointing things out for improvement and/or giving you the "outside perspective".